t/fits.t

```diff
@@ -2,340 +2,480 @@ use strict;
+
+        my $ar = sequence( 10, 10, 10 );
+        my $ai = zeroes($ar);
+        fftnd( $ar, $ai );
+        my ( $fh, $file ) = tfile;
+        wfits( $ar, $file );
+        my $y = rfits($file);
+        is_pdl $ar, $y,
+          "fftnd output (non-contiguous in memory) is written correctly";
+    }
+};
 
 ##############################
 # Check multi-HDU read/write
 
-my $x = sequence(5,5);
-my $y = rvals(5,5);
-
-our @aa;
+subtest 'multi-HDU read/write' => sub {
+    my $x = sequence( 5, 5 );
+    my $y = rvals( 5, 5 );
 
-lives_ok { wfits([$x,$y],$file) } "wfits with multiple HDUs didn't fail";
+    my @aa;
 
-lives_ok { @aa = rfits($file) } "rfits in list context didn't fail";
+    my ( $fh, $file ) = tfile;
+    lives_ok { wfits( [ $x, $y ], $file ) }
+    "wfits with multiple HDUs didn't fail";
 
-is_pdl $aa[0], $x, "first element reproduces written one";
-is_pdl $aa[1], $y, "Second element reproduces written one";
+    lives_ok { @aa = rfits($file) } "rfits in list context didn't fail";
 
-unlink $file;
+    is_pdl $aa[0], $x, "first element reproduces written one";
+    is_pdl $aa[1], $y, "Second element reproduces written one";
+};
 
 ##############################
 # Rudimentary check for longlong support
-SKIP:{
-	eval "use PDL::Types";
-	our $PDL_LL;
-    	skip "Longlong not supported",5   unless ($PDL_LL//0);
-
-	$x = rvals(longlong,7,7);
-	eval { wfits($x, $file); };
-	is $@, '', "writing a longlong image succeeded";
-	eval { $y = rfits($file); };
-	is $@, '', "Reading the longlong image succeeded";
-	isa_ok $y->hdr, "HASH", "Reading the longlong image produced a PDL with a hash header";
-	is $y->hdr->{BITPIX}, 64, "BITPIX value was correct";
-	is_pdl $y, $x, "The new image matches the old one (longlong)";
-	unlink $file;
-}
+subtest 'longlong image' => sub {
+    my ( $fh, $file ) = tfile;
+
+    my $x = rvals( longlong, 7, 7 );
+    lives_ok { wfits( $x, $file ) } 'write';
+    my $y;
+    lives_ok { $y = rfits($file); } 'read';
+    isa_ok $y->hdr, "HASH", "header is hash";
+    is $y->hdr->{BITPIX}, 64, "BITPIX value was correct";
+    is_pdl $y, $x, "The new image matches the old one";
+
+};
 
 ###############################
 # Check that tilde expansion works
 
-my $tildefile = cfile('~',"PDL-IO-FITS-test_$$.fits");
+subtest 'tilde path expansion' => sub {
 
 # Only read/write the tildefile if the directory is writable.
 # Some build environments, like the Debian pbuilder chroots, use a non-existent $HOME.
 # See: https://github.com/PDLPorters/pdl/issues/238
-if(-w dirname($tildefile)) {
-	lives_ok { sequence(3,5,2)->wfits($tildefile) } "wfits tilde expansion didn't fail";
-	lives_ok { rfits($tildefile) } "rfits tilde expansion didn't fail";
-	$tildefile =~ s/^(~)/glob($1)/e; #use the same trick as in FITS.pm to resolve this filename.
-	unlink($tildefile) or warn "Could not delete $tildefile: $!\n"; #clean up.
-}
+  SKIP: {
+        my $tilde_dir = <~>;
+
+        skip '~ directory is not writeable'
```

Comment:
This is an incorrect call to `skip`.

Reply:
Looks like Test2's skip doesn't require the count, which is why it passed on my end. fixed.